packages/fuel-gauge/src/transaction.test.ts

```diff
@@ -317,4 +318,46 @@ describe('Transaction', () => {
       })
     );
   });
+
+  it('should handle transaction with TestCoin', async () => {
+    using launched = await launchTestNode();
+    const { provider, wallets: [wallet] } = launched;
+
+    const baseAssetId = await provider.getBaseAssetId();
+    const initialBalance = await wallet.getBalance(baseAssetId);
+
+    // Create test coins with specific parameters
+    const testCoin = new TestCoin({
+      amount: bn(1000),
+      owner: wallet.address.toB256(),
+      assetId: baseAssetId,
+    });
```

Comment:
Is this test working? The transaction is being submitted with a coin not created on the local chain.